### PR TITLE
Add trailers to Status for BlockingResponseStream

### LINKED ISSUE #1861: Metadata trailers are not passed to Status*Exception when using BlockingResponseStream
Hi @nmittler,

thanks for improving the error handling via #1811 and solving #681.

When using blockingStubs on the client the metadata trailers are not passed into the Status*Exception. Your changeset adds this ability to 2 of 3 onClose methods of stub/ClientCalls.java.

Did you just miss to add the trailer handling to the BlockingResponseStream class in https://github.com/grpc/grpc-java/blob/641cb357c68cba5f0d1820973c6a6c0c908dc074/stub/src/main/java/io/grpc/stub/ClientCalls.java#L485 or is there another reason for dropping the trailers in this case?

Regards,
Björn

### PR DESCRIPTION
**Pull request by nmittler**
Fixes #1861

### CODE REVIEW

**Comment by nmittler** [comment]
@ejona86 PTAL

File: stub/src/main/java/io/grpc/stub/ClientCalls.java [hunk 1]
```diff
@@ -482,7 +485,7 @@ public void onClose(Status status, Metadata trailers) {
         if (status.isOk()) {
           buffer.add(BlockingResponseStream.this);
         } else {
-          buffer.add(status);
+          buffer.add(status.asRuntimeException(trailers));
```
Comment (ejona86) [inline stub/src/main/java/io/grpc/stub/ClientCalls.java]: Using an exception to pass the trailers is pretty heavy-weight. Could you swap to something that doesn't have to create a back trace?

**Comment by ejona86** [comment]
Please add/improve a test to verify the behavior.

**Comment by carl-mastrangelo** [comment]
@nmittler Eric and I are trying out a commit message scheme to make it easier to build the release notes.  The scheme is:

mainpackage: lowercase sentence fragment of what changed without a period

The sentence fragment should complete the sentence: "This change modifies grpc to _."

**Comment by nmittler** [comment]
@ejona86 PTAL

**Comment by ejona86** [comment]
@nmittler LGTM

**Comment by nmittler** [comment]
Cherry-picked as c102dd4

**Pull request closed without merging**